aws/resource_aws_api_gateway_integration.go

```diff
@@ -160,8 +172,8 @@ func resourceAwsApiGatewayIntegrationCreate(d *schema.ResourceData, meta interfa
 		RequestParameters:   aws.StringMap(parameters),
 		RequestTemplates:    aws.StringMap(templates),
 		Credentials:         credentials,
-		CacheNamespace:      nil,
-		CacheKeyParameters:  nil,
+		CacheNamespace:      aws.String(d.Get("resource_id").(string)),
```

Comment:
Also I think we should leave this as `nil` in case the user didn't specify any caching.